mlir/lib/Target/LLVMIR/DataLayoutImporter.cpp

```diff
@@ -197,9 +197,9 @@ DataLayoutImporter::tryToEmplaceStackAlignmentEntry(StringRef token) {
   if (failed(alignment))
     return failure();
 
-  // Only store the stack alignment if it has a non-default value.
+  // Stack alignment shouldn't be zero.
   if (*alignment == 0)
-    return success();
+    return failure();
```

Comment:
@gysit 
Please take a look if this change is ok.
Previously, "S0" in data layout string was equivalent to omitting it. After this patch, specifying zero for alignment is not allowed.

Reply:
Looks good to me!